cdap-app-templates/cdap-etl/hydrator-test/src/main/java/co/cask/cdap/etl/mock/condition/MockCondition.java

```diff
@@ -54,6 +56,29 @@ public MockCondition(Config config) {
   public boolean apply(ConditionContext context) throws Exception {
     String propertyName = config.name + ".branch.to.execute";
     String propertyValue = context.getArguments().get(propertyName);
+    // Check stage statistics
+    for (Map.Entry<String, StageStatistics> entry : context.getStageStatistics().entrySet()) {
+      String stageName = entry.getKey();
+      StageStatistics statistics = entry.getValue();
+      if (context.getArguments().has(stageName + ".stats")) {
```

Comment:
instead of checking here and throwing an exception if it's wrong, I think it's better to write out the statistics to the table, then read the table in the test case and check the value with assertEquals